lib/handlers.js

```diff
@@ -29,15 +29,19 @@ async function preValidationHandler (request) {
 }
 
 async function preExecutionHandler (request) {
-  const { errors, modifiedDocument } = await preExecutionHooksRunner(
+  const { errors, modifiedDocument, modifiedSchema } = await preExecutionHooksRunner(
     request.context.preExecution,
     request
   )
   if (errors.length > 0) {
     addErrorsToContext(request.context, errors)
   }
-  if (typeof modifiedDocument !== 'undefined') {
-    return { modifiedDocument, modifiedQuery: print(request.document) }
+  if (typeof modifiedDocument !== 'undefined' || typeof modifiedSchema !== 'undefined') {
+    return {
+      modifiedSchema,
+      modifiedDocument,
+      modifiedQuery: print(request.document)
```

Comment:
are we using modifiedQuery anywhere?

Reply:
It is used on the gateway implementation

Reply:
Maybe make it a getter?

Reply:
It was unused.
Removed

The same name is used on the `preGatewayExecutionHandler`